### ISSUE TITLE
[Breaking] Remove `[DataTestMethod]`

### ISSUE THREAD

**Issue by Evangelink** [microsoft/testfx#4166]
They are fully interchangeable and `[DataTestMethod]` as no different value so I suggest to mark it as obsolete (let's use a special diagnostic code to make it easy for people).

**Comment by nohwnd** [comment]
https://github.com/approvals/ApprovalTests.Net/blob/master/src/ApprovalTests/Namers/UnitTestFrameworks/MsTestDataTestMethodStackTraceParser.cs

Looks like there are some more usages of this as well. Are you sure this won't have impact that outweighs the benefit (are there any, other than having one less class?)

https://grep.app/search?q=DataTestMethodAttribute

**Comment by Evangelink** [comment]
It will definitely cause warnings but the class is fully pointless so I want to get rid of it in v4 meaning we need to give some transitioning time to users.

**Comment by SteveSyfuhs** [comment]
Yeah, this is a really annoying change for what appears to be zero value to the consumer. All it's doing is causing churn in test code. Please reconsider how much time and energy you're asking all consumers to invest with this change. It could be as simple as find and replace but it might not be.

**Comment by Evangelink** [comment]
Dropping the milestone for now but I still strongly believe this is just an anchor we should get rid of.

**Comment by Youssef1313** [comment]
Note, if we implemented analyzer/codefix as a first step, the codefix should handle the case where both `[TestMethod]` and `[DataTestMethod]` co-existed together. In that case, the DataTestMethod attribute should be removed completely and not replaced by `[TestMethod]`.

**Comment by Youssef1313** [comment]
Reverted in #6091, moving to v5.
